server/src/main/java/org/elasticsearch/indices/breaker/HierarchyCircuitBreakerService.java

```diff
@@ -334,4 +360,116 @@ private CircuitBreaker validateAndCreateBreaker(BreakerSettings breakerSettings)
                 this,
                 breakerSettings.getName());
     }
+
+    private static OverLimitStrategy createDoubleCheckStrategy(JvmInfo jvmInfo, LongSupplier currentMemoryUsageSupplier,
+                                                               LongSupplier timeSupplier, long minimumInterval) {
+        if (jvmInfo.useG1GC().equals("true")
+            // messing with GC is "dangerous" so we apply an escape hatch. Not intended to be used.
+            && Boolean.parseBoolean(System.getProperty("es.real_memory_circuit_breaker.g1.double_check.enabled", "true"))) {
```

Comment:
Do you mind using `Booleans.parseBoolean(System.getProperty("es.real_memory_circuit_breaker.g1.double_check.enabled"), true)`? The java Boolean parsing is pretty lenient and I thought it was a forbidden api at some point :|

Reply:
Thanks, fixed in 845dc109955ad5dd2b60f247541de9c0fc328225